Fix crashes due to missing TrueHD overrun checks

anssih opened the pull request:
Backport of #16813, for issue #16704.

CAEBitstreamPacker and CDVDAudioCodecPassthrough copy TrueHD frames into
buffers without checking for overruns.

Fix that.

However, audio dropouts still remain. To fix those, the TrueHD packet
offsets will need to be dynamically scaled based on the input_timing
field (TrueHD bytes 2-3) which allows determining the amount of time
required between the previous frame and the current frame.

(cherry picked from commit 0e76bd058fbd8e3d75e577c235f1e3b148f75246)

anssih committed:
Fix crashes due to missing TrueHD overrun checks

CAEBitstreamPacker and CDVDAudioCodecPassthrough copy TrueHD frames into
buffers without checking for overruns.

Fix that.

This should fix crashes of issue #16704.

However, audio dropouts still remain. To fix those, the TrueHD packet
offsets will need to be dynamically scaled based on the input_timing
field (TrueHD bytes 2-3) which allows determining the amount of time
required between the previous frame and the current frame.

(cherry picked from commit 0e76bd058fbd8e3d75e577c235f1e3b148f75246)

fritsch commented:
Code is fine. Minor clang-format warning, see: https://jenkins.kodi.tv/job/BuildMulti-PR/14127/artifact/PR17068-0001-de114b7193-Fix-crashes-due-to-missing-TrueHD-.diff

fritsch reviewed (changes requested):
Code does fine, please check the clang formating according to our pretty new coding guideline. Thanks very much.

anssih committed:
Fix cosmetic style issues introduced by TrueHD fix

Commit 0e76bd058fbd8e3 ("Fix crashes due to missing TrueHD overrun
checks") introduced several style issues (too long lines and missing
whitespace).

Fix those.

(cherry picked from commit c712930de281d8caaf94a1c8713b7106185642b2)
